client/state/directly/reducer.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	DIRECTLY_INITIALIZE
+} from 'state/action-types';
+import { createReducer } from 'state/utils';
+
+export const isInitialized = createReducer( false, {
+	[ DIRECTLY_INITIALIZE ]: () => true,
+} );
```

Comment:
>@aduth should chime in here since he's the wizard behind it.

Nope 😛  (Originally #4805)

It's a little unfair to claim that the utility creates heavy reducers though; [by it's implementation](https://github.com/Automattic/wp-calypso/blob/b2740526ef8f88e3850fdfc9078c93d5f0ff3fbe/client/state/utils.js#L214-L227), it's effectively a `hasOwnProperty` check for an unhandled action type.

The main advantage I see, aside from compactness, is turning state persistence into an opt-in than opt-out by default.